File: src/main/java/com/freya02/botcommands/application/slash/impl/GlobalSlashEventImpl.java
package com.freya02.botcommands.application.slash.impl;

import com.freya02.botcommands.BContext;
import com.freya02.botcommands.application.slash.GlobalSlashEvent;
import net.dv8tion.jda.api.events.interaction.commands.SlashCommandEvent;
import net.dv8tion.jda.internal.interactions.commands.SlashCommandInteractionImpl;

public class GlobalSlashEventImpl extends GlobalSlashEvent {
	private final BContext context;

	public GlobalSlashEventImpl(BContext context, SlashCommandEvent event) {
		super(event.getJDA(), event.getResponseNumber(), (SlashCommandInteractionImpl) event.getInteraction());
		
		this.context = context;
	}

	public BContext getContext() {
		return context;
	}
}

File: src/main/kotlin/com/freya02/botcommands/internal/commands/application/slash/GlobalSlashEventImpl.kt
package com.freya02.botcommands.internal.commands.application.slash

import com.freya02.botcommands.api.commands.application.slash.GlobalSlashEvent
import com.freya02.botcommands.api.commands.ratelimit.CancellableRateLimit
import com.freya02.botcommands.api.core.BContext
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent

internal open class GlobalSlashEventImpl(
    context: BContext,
    event: SlashCommandInteractionEvent,
    cancellableRateLimit: CancellableRateLimit
) : GlobalSlashEvent(context, event, cancellableRateLimit)